### PR TITLE
[Un-revert][Glibc] Configure modulemap for target, not host

### PR DESCRIPTION
**Pull request by modocache**
#### What's in this pull request?

This reverts commit f2154ee94d, which reverted 04e1cd5bda. The original commit needed to be reverted because of an issue in which install targets were added to OS X builds that did not target Linux. This addresses that issue by guarding all the Linux-only CMake logic with a check for the SDK being built.

/cc @gribozavr @jrose-apple 
#### Resolved bug number: None

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: stdlib/public/Glibc/CMakeLists.txt [hunk 1]
```diff
@@ -3,44 +3,57 @@ set(sources
     Misc.c
 )
 
-set(output_dir "${SWIFTLIB_DIR}/glibc")
+# When cross-compiling the stdlib on Unix platforms, we'll need a separate
+# glibc for each target.
+foreach(SDK ${SWIFT_SDKS})
+  foreach(arch ${SWIFT_SDK_${SDK}_ARCHITECTURES})
+    # Don't generate Glibc module maps for Darwin targets.
+    if("${SDK}" STREQUAL "LINUX" OR "${SDK}" STREQUAL "FREEBSD")
```
Comment (modocache) [inline stdlib/public/Glibc/CMakeLists.txt]: This `if()` statement is the only difference between this and https://github.com/apple/swift/pull/1679. :point_up:

**Comment by modocache** [comment]
@swift-ci wasn't able to catch the failure in https://github.com/apple/swift/pull/1679 because it doesn't trigger a build that uses `--install-swift=1`. I just ran `utils/build-toolchain` locally (which uses that parameter) with this patch, and everything worked! :100:

**Comment by gribozavr** [comment]
@swift-ci Please test

**Comment by modocache** [comment]
@gribozavr Looks like the tests passed! Is this good to merge?

**Comment by gribozavr** [comment]
Yes, LGTM, thanks @modocache!

**Comment by modocache** [comment]
Thanks! :wave:

**Comment by najacque** [comment]
This change looks like it might be causing the package job to fail:

/tmp/swift-package-tests/test-multi-compile-glibc/Output/swift-multi-compile-glibc.py.tmp.dir/hello.swift:1:8: error: missing required module 'SwiftGlibc'
import Glibc
 
Could you please take a look:  

https://ci.swift.org/view/Packages/job/oss-swift-package-linux-ubuntu-14_04/800/console
https://ci.swift.org/view/Packages/job/oss-swift-package-linux-ubuntu-15_10/792/console

**Comment by modocache** [comment]
Thanks for the heads up, @najacque. The failing jobs are using the `buildbot_linux_1404` and `buildbot_linux_1510` build script presets, which include the `--test-installable-package` flag. That flag isn't included in `utils/build-toolchain`, which I used locally to verify this pull request was OK. That explains why I missed this problem.

I'm looking into why the tests are failing now. The changes in #1757 seem relevant to this discussion as well...

**Comment by modocache** [comment]
@najacque I just finished a clean build of the `buildbot_linux_1510` preset on an Ubuntu 15.10 machine, using the tip of `master` from about an hour ago (b094e776b559ae2e6b40314743404cc604ac2b30), and everything passed, including the package tests:

```
+ cd /home/modocache/GitHub/apple/swift-integration-tests
+ python /home/modocache/GitHub/apple/llvm/utils/lit/lit.py . -sv --param package-path=/home/modocache/GitHub/apple/build/buildbot_linux/none-swift_package_sandbox --param filecheck=/home/modocache/GitHub/apple/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck
lit.py: lit.cfg:101: note: 'pexpect' module unavailable, skipping related tests
lit.py: lit.cfg:117: note: testing package: '/home/modocache/GitHub/apple/build/buildbot_linux/none-swift_package_sandbox'
lit.py: lit.cfg:144: note: testing using 'FileCheck': '/home/modocache/GitHub/apple/build/buildbot_linux/llvm-linux-x86_64/bin/FileCheck'
lit.py: lit.cfg:147: note: testing using 'swift': '/home/modocache/GitHub/apple/build/buildbot_linux/none-swift_package_sandbox/usr/bin/swift'
lit.py: lit.cfg:150: note: testing using 'swiftc': '/home/modocache/GitHub/apple/build/buildbot_linux/none-swift_package_sandbox/usr/bin/swiftc'
lit.py: lit.cfg:153: note: testing using 'lldb': /home/modocache/GitHub/apple/build/buildbot_linux/none-swift_package_sandbox/usr/bin/lldb
Testing Time: 9.80s
  Expected Passes    : 11
  Unsupported Tests  : 5
```

I wonder if maybe this problem is now fixed, perhaps thanks to #1757. I'll do a bit of bisecting. In the meantime, could you confirm:

- Whether this problem is now resolved?
- If it is now resolved, is any work necessary to backport a fix, such as picking a fix into a snapshot release branch or somerhing?

File: stdlib/public/Glibc/CMakeLists.txt [hunk 2]
```diff
@@ -3,44 +3,57 @@ set(sources
...
+          OUTPUT "${output_dir}/glibc.modulemap" "${output_dir}"
+          DEPENDS "${modulemap_path}"
+          COMMENT "Copying Glibc module to ${output_dir}")
 
-swift_install_in_component(stdlib
-    FILES "${output_dir}/module.map"
-    DESTINATION "lib/swift/glibc")
+      swift_install_in_component(stdlib
+          FILES "${output_dir}/glibc.modulemap"
+          DESTINATION "${output_dir}")
```
Comment (rintaro) [inline stdlib/public/Glibc/CMakeLists.txt]: I think, this line was the problem.
`${SWIFTLIB_DIR}/${SWIFT_SDK_${SDK}_LIB_SUBDIR}/${arch}`
is not relevant for install destination.

**Comment by modocache** [comment]
I just tried checking out the merge commit for this pull request, 38ceced77984188b25af9673ea61b28dc4c90b60, and running the package tests. Unfortunately, that worked as well--no test failures like the ones that appear in the CI build logs... there must be something else I've overlooking. I'll continue looking into this, sorry for the trouble.

Ideas on what might be failing would be appreciated! :bow:

**Comment by najacque** [comment]
@shahmishal can you comment on the current state of the tests?

**Comment by shahmishal** [comment]
@najacque @modocache Build passed :) Thanks! 
https://ci.swift.org/view/Packages/job/oss-swift-package-linux-ubuntu-14_04/

**Comment by modocache** [comment]
Oops, never mind--it was a build artifact that caused 38ceced to pass for me locally. That commit *does* fail with the same failures you've been seeing @najacque. So I do think that this has since been resolved.

File: stdlib/public/Glibc/CMakeLists.txt (hunk 2 above)
Comment (modocache) [inline stdlib/public/Glibc/CMakeLists.txt]: Awesome, thanks @rintaro! I guess your removal of this directive in #1757 resolved the issue.

**Comment by najacque** [comment]
Ok, thanks for looking into it!

**Pull request merged**